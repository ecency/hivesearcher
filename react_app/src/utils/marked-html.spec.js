/* eslint-disable */
import markedHtml from './marked-html';

describe('Marked Html', () => {
    it('(1) Remove all html but not <mark>', () => {
        let input = '<strong>lorem</strong> ipsum dolor sit <mark>amet</mark>';
        expect(markedHtml(input)).toMatchSnapshot();
    });

    it('(2) Should trim and remove multiple spaces', () => {
        let input = 'Etiam iaculis quis diam nec elementum.         Nulla vel facilisis massa.     Sed eu.        ';
        expect(markedHtml(input)).toMatchSnapshot();
    });

    it('(3) Should truncate', () => {
        let input = 'Etiam iaculis quis diam nec elementum.';
        expect(markedHtml(input, 10)).toMatchSnapshot();
    });

    it('(4) Should remove markdown', () => {
        let input = "# Lorem **ipsum** dolor sit amet, consectetur adipiscing elit. Quisque [hendrerit](http://google.com) vulputate ligula, id interdum metus volutpat id. <code>Vivamus</code> eget _congue_ nisi, at mollis ante.";
        expect(markedHtml(input)).toMatchSnapshot();
    });
});