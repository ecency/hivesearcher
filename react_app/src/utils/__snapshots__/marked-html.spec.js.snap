// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marked Html (1) Remove all html but not <mark> 1`] = `"lorem ipsum dolor sit <mark>amet</mark>"`;

exports[`Marked Html (2) Should trim and remove multiple spaces 1`] = `"Etiam iaculis quis diam nec elementum. Nulla vel facilisis massa. Sed eu."`;

exports[`Marked Html (3) Should truncate 1`] = `"Etiam iacu"`;

exports[`Marked Html (4) Should remove markdown 1`] = `"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque hendrerit vulputate ligula, id interdum metus volutpat id. Vivamus eget congue nisi, at mollis ante."`;
